src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs

```diff
@@ -51,5 +55,27 @@ public async Task DisposeAsync()
             await _lifetimeManager.DisposeAsync();
             (ServiceProvider as IDisposable)?.Dispose();
         }
+
+        IInternalServiceHubContext IInternalServiceHubContext.WithEndpoints(IEnumerable<ServiceEndpoint> endpoints)
+        {
+            var targetEndpoints = _endpointManager.GetEndpoints(_hubName).Intersect(endpoints, EqualityComparer<ServiceEndpoint>.Default).Select(e => e as HubServiceEndpoint).ToList();
```

Comment:
endpoint is not neccessorily a `HubServiceEndpoint`, actually user can never create a `HubServiceEndpoint`, is `WithEndpoint` to be exposed to the customer?

Reply:
The parameter 'endpoints' doesn't need to be `HubServiceEndpoint`. In the `ServiceHubContextFacts`, a test to demonstrate such situation where `ServiceEndpoint` is passed. As `_endpointManager.GetEndpoints(_hubName)` returns `IEnumerable<HubServiceEndpoint`, the intersect result will be `IEnumerable<HubServiceEndpoint>`. The `EqualityComparer` is `EqualityComparer<ServiceEndpoint>.Default`, which will use the `ServiceEndpoint.Equals(...)` and `ServiceEndpoints.GetHashCode()` to comparer.

Reply:
And `e as HubServiceEndpoint` will be always null for them?

Reply:
No, it always works. ` _endpointManager.GetEndpoints(_hubName).Intersect(...)`   returns endpoints in endpoint manager.